@3rdweb-sdk/react/hooks/useRegistry.ts

```diff
@@ -1,18 +1,96 @@
-import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useMutation, useQueryClient, useQuery } from "@tanstack/react-query";
 import { useAddress, useSDK, useSigner } from "@thirdweb-dev/react";
 import {
   addContractToMultiChainRegistry,
   getGaslessPolygonSDK,
 } from "components/contract-components/utils";
 import { useAllChainsData } from "hooks/chains/allChains";
-import { getDashboardChainRpc } from "lib/rpc";
-import { getEVMThirdwebSDK } from "lib/sdk";
 import invariant from "tiny-invariant";
+import {
+  useSupportedChains,
+  useSupportedChainsRecord,
+} from "hooks/chains/configureChains";
+import { useMemo } from "react";
+import { getAllMultichainRegistry } from "dashboard-extensions/common/read/getAllMultichainRegistry";
+import { polygon } from "thirdweb/chains";
+import { getContract } from "thirdweb";
+import { thirdwebClient } from "lib/thirdweb-client";
+import { BasicContract } from "contract-ui/types/types";
+
+export const MULTICHAIN_REGISTRY_ADDRESS = "0xcdAD8FA86e18538aC207872E8ff3536501431B73";
+
+export function useMultiChainRegContractList(walletAddress?: string) {
+  return useQuery(
+    ["dashboard-registry", walletAddress, "multichain-contract-list"],
+    async () => {
+      invariant(walletAddress, "walletAddress is required");
+      const contract = getContract({
```

Comment:
since this is static define it *outside* of the function, it cannot change -> just define it once and use it inside